js/utils/utils.js

```diff
@@ -0,0 +1,116 @@
+const shaLib = require('./sha256.js');
+const ripemd160Lib = require('./ripemd160.js');
+
+module.exports = {
+  serializeHex: (uint8arr) => {
+    if (!uint8arr) {
+      return '';
+    }
+
+    let hexStr = '';
+    for (let i = 0; i < uint8arr.length; i++) {
+      let hex = (uint8arr[i] & 0xff).toString(16);
+      hex = (hex.length === 1) ? '0' + hex : hex;
+      hexStr += hex;
+    }
+
+    return `0x${hexStr.toLowerCase()}`;
+  },
+
+  deserializeHex: (hexStr) => {
+    if (!hexStr) {
+      return new Uint8Array();
+    }
+
+    let hex = ''
+    if (hexStr.slice(0, 2) === '0x') {
+      hex = hexStr.slice(2);
+    } else {
+      hex = hexStr;
+    }
+
+    let a = [];
+    for (let i = 0; i < hex.length; i+=2) {
+      a.push(parseInt(hex.substr(i,2),16));
+    }
+
+    return new Uint8Array(a);
+  },
+
+  sha256: (buf) => {
+    return module.exports.deserializeHex(shaLib(buf));
+  },
+
+  ripemd160: (buf) => {
+    return ripemd160Lib.default(buf);
+  },
+
+  typedArraysAreEqual: (a, b) => {
+    if (a.byteLength !== b.byteLength) return false;
+    return a.every((val, i) => val === b[i]);
```

Comment:
I'm not sure what the metering model would be, but you could save gas by returning early here when a single element doesn't match instead of using `every`

Reply:
ah you're probably right. something like this:

```
if (a.byteLength !== b.byteLength) return false;
if (a.BYTES_PER_ELEMENT !== b.BYTES_PER_ELEMENT) return false;
for (let i = 0; i < a.byteLength; i += 1) {
  if (a[i] != b[i]) return false;
}
return true;
```

Reply:
Updated